### PR TITLE
Introduce bir text format

### LINKED ISSUE #1031: Implement bir textual format
Design a textual format to represent bir. Implement a serializer and a deserializer for it.

### PR DESCRIPTION
**Pull request by manuranga**
Part of #1031

### CODE REVIEW

**Comment by manuranga** [comment]
Implements serialization, will send deserialization separately.

File: compiler/main.bal [hunk 1]
```diff
@@ -17,6 +17,7 @@ public type Options record {
     string? expectOutDir = ();
     string? gc = ();
     string? htmlError = ();
+    boolean outBir = false;
```
Comment (jclark) [inline compiler/main.bal]: We can't have `outBir` and `outDir`. Too lexically close but semantically different.

File: compiler/compile.bal [hunk 2]
```diff
@@ -18,33 +19,72 @@ type Job record {|
     ProcessedImport|JOB_IN_PROGRESS? result;
 |};
 
-class CompileContext {
+public type Emitter object {
```
Comment (jclark) [inline compiler/compile.bal]: This shouldn't be public

**Comment by jclark** [comment]
Let's fix the minor nits, and then land this.

File: compiler/compile.bal (hunk 2 above)
Comment (manuranga) [inline compiler/compile.bal]: make module level.

**Review by jclark — DISMISSED** [review]

**Review by jclark — APPROVED** [review]

**Pull request merged**